dingo/gw/dataset/waveform_dataset.py

```diff
@@ -210,20 +240,67 @@ def __getitem__(self, idx) -> Dict[str, Dict[str, Union[float, np.ndarray]]]:
         return self.__getitems__([idx])[0]
 
     def __getitems__(
-        self, batched_idx
-    ) -> Dict[str, Dict[str, Union[float, np.ndarray]]]:
+        self, batched_idx: list
+    ) -> List[Dict[str, Dict[str, Union[float, np.ndarray]]]]:
         """
         Return a nested dictionary containing parameters and waveform polarizations
         for sample with index `idx`. If defined, a chain of transformations is applied to
         the waveform data.
         """
-        parameters = {
-            k: v if isinstance(v, float) else v.to_numpy()
-            for k, v in self.parameters.iloc[batched_idx].items()
-        }
-        polarizations = {
-            pol: waveforms[batched_idx] for pol, waveforms in self.polarizations.items()
-        }
+
+        # Get parameters and data for idx
+        if self.leave_polarizations_on_disk:
+            # Load polarizations from disk
+            if self.file_handle is None:
+                # Open hdf5 file
+                self.file_handle = h5py.File(self.file_name, "r")
+
+            polarizations = recursive_hdf5_load(
+                self.file_handle, keys=["polarizations"], idx=batched_idx
+            )["polarizations"]
+            # Apply domain update to set waveform to zero for f < f_min
+            if self.svd is None:
+                polarizations = {
+                    pol: self.domain.update_data(waveforms)
+                    for pol, waveforms in polarizations.items()
+                }
+            parameters = {
+                k: v if isinstance(v, float) else v.to_numpy()
+                for k, v in self.parameters.iloc[batched_idx].items()
+            }
+            # Convert parameters to dict
+            if not isinstance(parameters, dict):
+                parameters = parameters.to_dict()
+            # Update precision
+            if (
+                self.precision is not None
+                and self.real_type is not None
+                and self.complex_type is not None
+            ):
+                parameters = {
+                    k: v.astype(self.real_type, copy=False)
+                    for k, v in parameters.items()
+                }
+                polarizations = {
+                    k: v.astype(self.complex_type, copy=False)
+                    for k, v in polarizations.items()
+                }
+            # Perform SVD size update on waveform
+            if self.svd is not None and self.svd_size_update is not None:
+                for k, v in polarizations.items():
+                    if len(v.shape) == 1:
+                        polarizations[k] = v[: self.svd_size_update]
+                    else:
+                        polarizations[k] = v[:, : self.svd_size_update]
+        else:
+            parameters = {
+                k: v if isinstance(v, float) else v.to_numpy()
```

Comment:
```suggestion
                k: v.to_numpy()
```
(This makes it consistent with #290.)

Reply:
Done.